core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,164 @@
+		} else if (!allDone) {
+			return false;
+		}
+		boolean shouldContinue = getEvaluatedCondition(task, iteration, workflow);
+		logger.debug("workflowid {} shouldcontinue {}",workflow.getWorkflowId(), shouldContinue);
+		if (shouldContinue){
+			return scheduleLoopTasks(task, iterations, workflow, provider);
+		} else {
+			return markLoopTaskSuccess(task, workflow, iterations);
+		}
+	}
+
+	Map<String, Object> getConditionInput(Task task, int iteration, Workflow workflow) {
+		Map<String, Object> map = new HashMap<>();
+		List<String> loopOverTask = task.getWorkflowTask().getLoopOver();
+		for (String taskName : loopOverTask) {
+			String refName = taskName;
+			if (iteration > 0) {
+				refName += "_" + (iteration);
+			}
+			Task temp = workflow.getTaskByRefName(refName);
+			map.putAll(temp.getOutputData());
+			logger.debug("Taskname {} output {}", temp.getReferenceTaskName(), temp.getOutputData());
+		}
+		return map;
+	}
+
+	boolean scheduleLoopTasks(Task task, Map<String, Integer> iterations, Workflow workflow, WorkflowExecutor provider) {
+		List<String> loopOver = (List<String>) task.getInputData().get("loopOver");
+		List<Task> taskToBeScheduled = new ArrayList<>();
+		int iteration = iterations.get(task.getTaskId()) == null ? 1 :  iterations.get(task.getTaskId()) + 1;
+		for(String loopOverRef : loopOver){
+			Task existingTask = workflow.getTaskByRefName(loopOverRef);
+			existingTask.setRetried(true);
+			Task newTask = provider.taskToBeRescheduled(existingTask);
+			newTask.setSeq(existingTask.getSeq());
+			newTask.setReferenceTaskName(existingTask.getReferenceTaskName() + "_" + iteration);
+			newTask.setRetryCount(existingTask.getRetryCount());
+			newTask.setScheduledTime(System.currentTimeMillis());
+			taskToBeScheduled.add(newTask);
+		}
+		iterations.put(task.getTaskId(), iteration);
+		provider.scheduleTask(workflow, taskToBeScheduled);
+		return false;
+	}
+
+	boolean markLoopTaskFailed(Task task, String failureReason) {
+		task.setReasonForIncompletion(failureReason);
+		task.setStatus(Status.FAILED);
+		logger.debug("taskid {} took {} iterations to failed",task.getTaskId(), iterations.get(task.getTaskId()));
+		iterations.remove(task.getTaskId());
+		return true;
+	}
+
+	boolean markLoopTaskSuccess(Task task, Workflow workflow, Map<String, Integer> iterations) {
+		logger.debug("workflowid {} took {} iterations to complete",workflow.getWorkflowId(), iterations.get(task.getTaskId()));
+		iterations.remove(task.getTaskId());
+		task.setStatus(Status.COMPLETED);
+		return true;
+	}
+
+	@VisibleForTesting
+	boolean getEvaluatedCondition(Task task,int iteration, Workflow workflow) {
+		Map<String, Object> taskInput = getConditionInput(task, iteration, workflow);
+		String condition = task.getWorkflowTask().getLoopCondition();
+		boolean shouldContinue = false;
+		if (condition != null) {
+			logger.debug("Case being evaluated using decision expression: {}", condition);
```

Comment:
Consider adding log for result of condition evaluation as well. Would be helpfull